app/scripts/TrackRenderer.jsx

```diff
@@ -85,6 +86,218 @@ const { getDataFetcher } = AVAILABLE_FOR_PLUGINS.dataFetchers;
 
 const SCROLL_TIMEOUT = 100;
 
+/** @typedef {TrackRenderer["setCenter"]} SetCentersFunction */
+/** @typedef {import("d3-scale").ScaleLinear<number, number>} ScaleLinear */
+/** @typedef {(x: ScaleLinear, y: ScaleLinear) => [ScaleLinear, ScaleLinear]} ProjectorFunction */
+/** @typedef {(xScale: ScaleLinear, yScale: ScaleLinear, k?: number, x?: number, y?: number, xPosition?: number, yPosition?: number) => void} ZoomedFunction */
+/** @typedef {'left' | 'right' | 'top' | 'bottom' | 'center' | 'gallery'} TrackPosition */
+
+/**
+ * @typedef TrackObject
+ * @property {() => void} draw
+ * @property {(options: unknown) => void} rerender
+ * @property {boolean} delayDrawing
+ * @property {Array<TrackObject>=} childTracks
+ * @property {Record<string, TrackObject>} createdTracks
+ * @property {(x: ScaleLinear, y: ScaleLinear) => void} refScalesChanged
+ * @property {[number, number]} position
+ * @property {[number, number]} dimensions
+ * @property {(contents: Array<TrackConfig>, x: unknown) => TrackObject} updateContents
+ * @property {ZoomedFunction} zoomed
+ * @property {(position: [number, number]) => void} setPosition
+ * @property {(position: [number, number]) => void} setDimensions
+ * @property {() => void} remove
+ * @property {(extent: number) => void} movedY
+ * @property {(yPosition: number, wheelDelta: number) => void} zoomedY
+ */
```

Comment:
I imagine that you arrived at these properties based on what properties of `TrackObject` were being used in `TrackRenderer`? I see that some of these properties are only used in specific`*Track` classes. For example, `delayDrawing` is used in `PixiTrack` but not `SVGTrack`.